src/message.rs

```diff
@@ -129,9 +129,24 @@ pub mod status {
 #[cfg_attr(feature = "defmt", derive(defmt::Format))]
 pub struct Channel(u8);
 
+impl Channel {
+    /// Create a new `Channel`
+    ///
+    /// # Arguments
+    /// * `val` - the 0 based channel value
+    ///
+    /// # Note
+    /// * The `val` will be clamped so it is in the 0..15 valid range.
+    ///
+    pub const fn new(val: u8) -> Self {
+        Self(if val > 15 { 15 } else { val })
```

Comment:
maybe use val.min(15) for this?

Reply:
`.min` isn't `const` and so i used `if` all over because of that

Reply:
Ah, should have figured that out myself.